packages/devtools_app/lib/src/service_manager.dart

```diff
@@ -564,24 +563,38 @@ class IsolateManager {
   }
 
   void _handleVmServiceClosed() {
+    cancel();
     _lastIsolateIndex = 0;
     _setSelectedIsolate(null);
     _isolateIndexMap.clear();
     _isolates.clear();
   }
+
+  void vmServiceOpened(VmServiceWrapper service) {
+    cancel();
+    _service = service;
+    autoDispose(service.onDebugEvent.listen(_handleDebugEvent));
+    autoDispose(service.onIsolateEvent.listen(_handleIsolateEvent));
+  }
 }
 
-class ServiceExtensionManager {
+class ServiceExtensionManager extends Disposer {
   VmServiceWrapper _service;
   IsolateManager _isolateManager;
 
-  bool _firstFrameEventReceived = false;
+  // Id so it is easy to avoid long running requests that were started
+  // for a previous vm service connection.
+  int vmServiceId = 0;
```

Comment:
Maybe I'm missing something, but I don't see where this is ever modified?

Reply:
nope that code was incomplete. Cleaned it up and switched to tracking the main isolate directly in the service manager as the list of service extensions is only valid for the main isolate anyway and some of the logic was just crazy if you did change isolates.